bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,471 @@
+                logger.warn("Exception: ", e);
+            }
+        }
+    }
+
+    public void updateChannel(String channelID, String channelValue) {
+        if (!channelValue.contains("null")) {
+            if (channelID.contains("consumption") || channelID.contains("Consumption")
+                    || channelID.contains("accumulatedProduction")) {
+                updateState(channelID, new QuantityType<>(new BigDecimal(channelValue), SmartHomeUnits.KILOWATT_HOUR));
+            } else if (channelID.contains("power") || channelID.contains("Power")) {
+                updateState(channelID, new QuantityType<>(new BigDecimal(channelValue), SmartHomeUnits.WATT));
+            } else if (channelID.contains("voltage")) {
+                updateState(channelID, new QuantityType<>(new BigDecimal(channelValue), SmartHomeUnits.VOLT));
+            } else if (channelID.contains("live_current")) {
+                updateState(channelID, new QuantityType<>(new BigDecimal(channelValue), SmartHomeUnits.AMPERE));
+            } else {
+                updateState(channelID, new DecimalType(channelValue));
+            }
+        }
+    }
+
+    public void thingStatusChanged(ThingStatusInfo thingStatusInfo) {
+        logger.debug("Thing Status updated to {} for device: {}", thingStatusInfo.getStatus(), getThing().getUID());
+        if (thingStatusInfo.getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Unable to communicate with Tibber API");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> pollingJob = this.pollingJob;
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            this.pollingJob = null;
+        }
+        if (isConnected()) {
+            close();
+        }
+        super.dispose();
+    }
+
+    public void open() throws Exception {
+        if (isConnected()) {
+            logger.debug("Open: connection is already open");
+        } else {
+            logger.info("Connecting to: {}", SUBSCRIPTION_URL);
+            TibberConfiguration tibberConfig = getConfigAs(TibberConfiguration.class);
+
+            sslContextFactory.setTrustAll(true);
+            sslContextFactory.setEndpointIdentificationAlgorithm(null);
+
+            WebSocketClient newClient = new WebSocketClient(sslContextFactory);
+            newClient.setMaxIdleTimeout(360 * 1000);
+
+            socket = new TibberWebSocketListener();
+
+            ClientUpgradeRequest newRequest = new ClientUpgradeRequest();
+
+            String token = new StringBuilder("Bearer ").append(tibberConfig.getToken()).toString();
+            newRequest.setHeader("Authorization", token);
+            newRequest.setSubProtocols("graphql-subscriptions");
+
+            newClient.start();
+            newClient.connect(socket, new URI(SUBSCRIPTION_URL), newRequest);
+        }
+    }
+
+    @SuppressWarnings("null")
```

Comment:
The code looks fine to me, why is this needed?

Reply:
If I remove this one, I get a warning within Eclipse IDE for the `socket` variable: "Potential null pointer access: This expression has @Nullable type". If I remove @Nullable from parameter initialisation, I get a warning for my `TibberHandler` method: "The @NonNull field socket may not have been initialised".

For the latest commit, I have removed `@SuppressWarnings` for this method, but have kept the `@Nullable` initialisation of the variable.